inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -19,15 +27,16 @@ def application_add(args: argparse.Namespace) -> str:
     manifest = ('<?xml version="1.0" encoding="utf-8"?>' +
                 '<manifest>' +
                 '<type>source</type>' +
-                '<applicationSource>' +
-                '<add><gpg>'
-                '{0}' +
-                '{1}'
-                '</gpg><repo><repos>').format(create_xml_tag(arguments, "uri"),
-                                              create_xml_tag(arguments, "keyname"))
+                '<applicationSource>' + '<add>')
+
+    if args.gpgKeyUri and args.gpgKeyName:
+        manifest += ('<gpg>' + '{0}' + '{1}' + '</gpg>').format(create_xml_tag(arguments, "uri"),
+                                                                create_xml_tag(arguments, "keyname"))
+
+    manifest += '<repo><repos>'
 
     for source in args.sources:
-        manifest += '<source_pkg>' + source.strip() + '</source_pkg>'
+        manifest += '<source_pkg>' + source + '</source_pkg>'
```

Comment:
Ensure that the `source_pkg` elements are properly sanitized to prevent potential security issues, such as command injection, if user input is not properly handled. [important]

Reply:
done